cli.py

```diff
@@ -8,28 +8,31 @@
 
 
 def main():
-    """Convert the given Doxygen index.xml to AsciiDoc and print the result."""
+    """Convert the given Doxygen index.xml to AsciiDoc and output the result."""
     parser = argparse.ArgumentParser()
     parser.add_argument("-f", "--file", help="The path of the file to convert", default=None)
+    parser.add_argument("-o", "--output", help="The path of the output file", default=None)
     parser.add_argument("-c", "--child", help="Is NOT the root index file", default=False, action='store_true')
     args = parser.parse_args()
     filename = args.file
+    output_filename = args.output
     is_child = args.child
     if filename:
         xmldir = os.path.dirname(filename)
         with open(filename, encoding="utf-8") as xml:
             if is_child:
-                print(
-                    Node(
-                        BeautifulSoup(xml, "xml").doxygen, xmldir=xmldir
-                    ).to_asciidoc()
-                )
+                result = Node(
+                    BeautifulSoup(xml, "xml").doxygen, xmldir=xmldir
+                ).to_asciidoc()
             else:
-                print(
-                    DoxygenindexNode(
-                        BeautifulSoup(xml, "xml").doxygenindex, xmldir=xmldir
-                    ).to_asciidoc()
-                )
-                
+                result = DoxygenindexNode(
+                    BeautifulSoup(xml, "xml").doxygenindex, xmldir=xmldir
+                ).to_asciidoc()
+        if output_filename is not None:
+            with open(output_filename, "w", encoding="utf-8") as output:
+                output.write(result)
+        else:
+            print(result)
```

Comment:
What happens if someone passes `-o` or `--output` with no argument? Does it come through as an empty string or as `None`?

If the former, would it be safer to use the inherit "truthiness" of a non-empty string like so:

```suggestion
        if output_filename:
            with open(output_filename, "w", encoding="utf-8") as output:
                output.write(result)
        else:
            print(result)
```

Reply:
> What happens if someone passes `-o` or `--output` with no argument?

Good question. I just did a quick test, and it automatically complains with `error: argument -o/--output: expected one argument` and prints the usage-string :smiley: